crates/chain-state/src/memory_overlay.rs

```diff
@@ -168,6 +171,17 @@ impl StateProofProvider for MemoryOverlayStateProvider {
         self.historical.proof(hashed_state, address, slots)
     }
 
+    // TODO: Currently this does not reuse available in-memory trie nodes.
```

Comment:
waiting on https://github.com/paradigmxyz/reth/pull/10929 to be merged, so we could remove the `TODO`